packages/next-app/src/app/metadata/[poolId]/page.tsx

```diff
@@ -1,35 +1,50 @@
 "use client";
 
-import { useRef, useState } from "react";
+import { useEffect, useRef, useState } from "react";
 
 import { Button } from "../../../components";
+import useDebounce from "../../../hooks/useDebounce";
 import metadataGql from "../../../lib/poolMetadataGql";
+import {
+  usePoolMetadataRegistrySetPoolMetadata,
+  usePreparePoolMetadataRegistrySetPoolMetadata
+} from "../../../wagmi/generated";
 
 export default function Page({ params }: { params: { poolId: string } }) {
```

Comment:
Something like ```params: { poolId: `0x${string}` };``` wouldn't raise typescript errors

Reply:
Good one! Thanks for the tip ;)